### PR TITLE
Refactoring code in Transaction and Account.

### PR DESCRIPTION
**Pull request by algonautshant**
1)
Account.java depends on Transaction.java
Many of the signing operations are in Account.java
Given the dependency direction, Transaction cannot utilize the signing operations in Account.
Moving the signing operations to Transaction, so that both Transaction and Account can use them.
Since some of the signing operations need the account, factored out the core signing operations to utils.SignatureUtils.

2)
Added SignatureUtils.getDummyPrivateKey utility to generate a private key and cache it.
This is used for signing transactions for the purpose of measuring the encoding size to determine the fee.
Note that a new key is not generated every time when getDummyPrivateKey is called.

3)
Added:
acceptAssetTransaction
assetTransferTransaction

4)
Reformatted indentation in Transaction.java
Added missing comments to Transaction.java

5)
Modified ApiException to include resoponseBody text with the message.

6)
Deprecated transactionWithSuggestedFeePerByte
replaced by Transaction.setFeeWithSuggestedFeePerByte

### CODE REVIEW

**Comment by pzbitskiy** [comment]
IMO this is questionable change. The idea of Java SDK is to have all signing operations inside Account class. Yes, it is different from other SDK's but it makes sense - in order to sign you need a secret key that is part of account class and is not accessible from outside.
I suggest another approach - have Transaction.bytesToSign() method and Account.signRawBytes or similar. So that we can preserve existing API and deprecate it.

File: src/main/java/com/algorand/algosdk/transaction/Transaction.java [hunk 1]
```diff
@@ -78,15 +122,219 @@
...
+     * unrelated to asset management)
+     * @param lastRound is the last round this txn is valid
+     * @param note
+     * @param genesisID corresponds to the id of the network
+     * @param genesisHash corresponds to the base64-encoded hash of the genesis
+     * of the network
+     * @param assetCreator is the address of the asset creator
+     * @param assetIndex is the asset index
+     **/
+    public static Transaction acceptAssetTransaction(Address acceptingAccount, //AssetTransaction
```
Comment (rotemh) [inline src/main/java/com/algorand/algosdk/transaction/Transaction.java]: Please consider naming it either by the verb `createAcceptAssetTransaction` or simply `acceptAsset`

File: src/main/java/com/algorand/algosdk/transaction/Transaction.java [hunk 2]
```diff
@@ -78,15 +122,219 @@
...
+     * unrelated to asset management)
+     * @param lastRound is the last round this txn is valid
+     * @param note
+     * @param genesisID corresponds to the id of the network
+     * @param genesisHash corresponds to the base64-encoded hash of the genesis
+     * of the network
+     * @param assetCreator is the address of the asset creator
+     * @param assetIndex is the asset index
+     **/
+    public static Transaction assetTransferTransaction(// AssetTransaction
```
Comment (rotemh) [inline src/main/java/com/algorand/algosdk/transaction/Transaction.java]: same here

File: src/main/java/com/algorand/algosdk/util/SignatureUtils.java [hunk 3]
```diff
@@ -0,0 +1,53 @@
+package com.algorand.algosdk.util;
```
[the remark this answers is not in the document]
Comment (rotemh) [reply src/main/java/com/algorand/algosdk/util/SignatureUtils.java]: rawSign and gen SK should already exist in the crypto lib. 
As @pzbitskiy mentioned, the additional `genDummy` should go there as well.

[the remark this answers is not in the document]
Comment (pzbitskiy) [reply src/main/java/com/algorand/algosdk/util/SignatureUtils.java]: > rawSign and gen SK should already exist in the crypto lib.
Surprisingly there is no signing functionality in `crypto`, only in `account`. Probably that's the reason why Java SDK looks so different from other SDKs.

**Comment by algonautshant** [comment]
> Some minor comments. General one - Why do we keep adding more constructors and not crating other classes that inherit from `Transaction`?

Because that will require a custom encoder/decoder.

**Comment by rotemh** [comment]
> > Some minor comments. General one - Why do we keep adding more constructors and not crating other classes that inherit from `Transaction`?
> 
> Because that will require a custom encoder/decoder.

Since all transaction encodings are flat, an alternative solution to avoid creating custom encode/decoder, is to create a function that will produce a serializable transaction object.

**Pull request closed without merging**